doc/migration/version-upgrade-guide.md

```diff
@@ -1068,13 +1068,13 @@ The methods `ctx.GetOperatorNamespace()` and `ctx.GetWatchNamespace()` was added
 
 This release contains breaking changes in some commands.
 
-- The --namespace flag from `operator-sdk run --local` command, `operator-sdk test --local` command and `operator-sdk cleanup` command was deprecated and was replaced by --watch-namespace and --operator-namespace .
+- The --namespace flag from `operator-sdk run --local` command, `operator-sdk test --local` command and `operator-sdk cleanup` command was deprecated and was replaced by --watch-namespace and --operator-namespace. [#2617](https://github.com/operator-framework/operator-sdk/pull/2617)
 
 Note that --watch-namespace can be used to set the namespace(s) which the operator will watch for changes. It will set the environment variable WATCH_NAMESPACE. Use explicitly an empty string to watch all namespaces or inform a List of namespaces such as "ns1,ns2" when the operator is cluster-scoped. If you use a List, then it needs contains the namespace where the operator is "deployed" in since the default metrics implementation will manage resources in the Operator's namespace. By default, it will be the Operator Namespace.
 
 Then, use the flag --operator-namespace to inform the namespace where the Operator will be "deployed" in and then, it will set the environment variable OPERATOR_NAMESPACE. If this value is not set, then it will be the namespace defined as default in the Kubeconfig.
 
-NOTE: For more information check the PRs which are responsible for the above changes [#2617](https://github.com/operator-framework/operator-sdk/pull/2617).
+- If you've run `operator-sdk bundle create --generate-only`, move your bundle Dockerfile at `<project-root>/deploy/olm-catalog/<operator-name>/Dockerfile` to `<project-root>/bundle.Dockerfile` and update the first `COPY` from `COPY /*.yaml manifests/` to `COPY deploy/olm-catalog/<operator-name>/<bundle-dir> manifests/`. [#2715](https://github.com/operator-framework/operator-sdk/pull/2715)
```

Comment:
@kevinrizza the v1.6 update breaks `--generate-only` behavior, since the argument to `--directory`, which is for a named bundle dir ex `deploy/olm-catalog/my-operator/0.0.1`, is placed in a top-level `bundle.Dockerfile`. Running `operator-sdk bundle create --directory deploy/olm-catalog/my-operator/0.0.2` after generating a metadata/Dockerfile for `0.0.1` results in a bundle for `0.0.1`.

How does `opm` work with a directory structure like this?

Reply:
we had a weird implementation where the `metadata/` folder was getting inserted *inside* the manifest folder and then the dockerfile had to extract it out. we modified it so that the scaffold matched what actually appeared on the directory structure in the container image.

that was one of the big purposes of this update. You shouldn't be creating versioned folders, instead you should have a single `manifests/` folder and a matching `metadata/` folder. the registry api update pushes the `metadata/` folder to the same directory level as the `manifest/` folder

Reply:
So users are expected to use a VCS to manage bundle versions going forward, and everything in `manifests/` is always latest?

@joelanford @jmccormick2001 @dmesser we're going to have to make a serious breaking change to the CSV generator/dir structure unless we can come up with a middle ground.

Reply:
Yes, this is what i was trying to say for a while. The CI pipelines are going to be built around having just one bundle in your repository and when you want to push a change you just update that bundle -- this is for a few reasons, but the big one is because the operator bundles are just single versions, and if we didn't do it this way we would have to come up with a bunch of complex CI that would need to know that a new folder was generated in a specific subdirectory and know how to aggregate it. Also, the `annotations.yaml` file is aggregated at the specific bundle version and not at the package layer.

Reply:
If the CI is looking for them in other location, so would not be possible to add flags there to pass the current location and the CI  job copy the files for anyplace that it likes?

Reply:
@kevinrizza #2746 helps users migrate to a `manifests/` format while supporting versioned bundle dirs. Next is for `generate csv` to do the same.